src/main/java/org/gridsuite/voltageinit/server/computation/service/AbstractWorkerService.java

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.voltageinit.server.computation.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.commons.reporter.ReporterModel;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.VariantManagerConstants;
+import com.powsybl.network.store.client.NetworkStoreService;
+import com.powsybl.network.store.client.PreloadingStrategy;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+/**
+ * @author Mathieu Deharbe <mathieu.deharbe at rte-france.com>
+ * @param <S> powsybl Result class specific to the computation
+ * @param <R> Run context specific to a computation, including parameters
+ * @param <P> powsybl and gridsuite Parameters specifics to the computation
+ * @param <T> result service specific to the computation
+ */
+public abstract class AbstractWorkerService<S, R extends AbstractComputationRunContext<P>, P, T extends AbstractComputationResultService<?>> {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorkerService.class);
```

Comment:
The logger is always unique to the class
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorkerService.class);
```

Reply:
Why ? 

For some other computation microservices we used the LOGGER of their abstract parent class and it works fine. Do you want to redeclare an other LOGGER in the child class ?

Reply:
Might be cleaner yes, to have the logger of the concrete class directly

Reply:
> Why ?
> For some other computation microservices we used the LOGGER of their abstract parent class and it works fine. Do you want to redeclare an other LOGGER in the child class ?

Yes, because a logger is normally used only by his class. In itself it work yes, but in the logs (console or file), you would see the logs coming from the abstract class and not from the child class, where it really originate.

If you want the child class to use the logger of the parent, then the abstract class init the logger with the child/true class of the instance:
```java
protected final Logger LOGGER = LoggerFactory.getLogger(this.getClass());
```